### PR TITLE
Code changes for Email attachment  as download and email attachmentr as link

### PR DESCRIPTION
**Pull request by premchandpl**
Code changes for Email attachment  as download and email attachmentr as link

### CODE REVIEW

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Taxonomy.cs [hunk 1]
```diff
@@ -394,11 +394,11 @@ private string GetManagedTermSetHierarchy(ClientContext clientContext, TermSet t
                                 jw.WritePropertyName("isNoFolderStructurePresent");
                                 if (generalSettings.IsBackwardCompatible)
                                 {
-                                    if(customProperty.Value==ServiceConstants.IS_FOLDER_STRUCTURE_PRESENT_FALSE)
+                                    if(customProperty.Value.ToLower()==ServiceConstants.IS_FOLDER_STRUCTURE_PRESENT_FALSE.ToLower())
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Taxonomy.cs]: Modified the code to do case insensitive comparison

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/DocumentProvision.cs [hunk 2]
```diff
@@ -142,22 +145,13 @@ private Stream GenerateEmail(Dictionary<string, Stream> collectionOfAttachments,
             Stream result = null;
             try
             {
+                
+                
+
+
                 MemoryStream mailFile = GetMailAsStream(collectionOfAttachments, documentUrls, attachmentFlag);
                 mailFile.Position = 0;
-                var fileContentResponse = new HttpResponseMessage(HttpStatusCode.OK);
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/DocumentProvision.cs]: Code has been moved to Controller

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/DocumentProvision.cs [hunk 3]
```diff
@@ -241,7 +235,7 @@ internal MemoryStream GetMailAsStream(Dictionary<string, Stream> collectionOfAtt
         /// </summary>
         /// <param name="fileExtension">Extension of the file</param>
         /// <returns>File content type</returns>
-        internal static string ReturnExtension(string fileExtension)
+        public static string ReturnExtension(string fileExtension)
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/DocumentProvision.cs]: Changed the access modifier

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs [hunk 4]
```diff
@@ -95,86 +102,35 @@ public IActionResult DownloadAttachmentsAsStream([FromBody]MailAttachmentDetails
...
-        /// For document attachment
-        /// </summary>
-        /// <param name="mailAttachmentDetails"></param>
-        /// <returns></returns>
-        [HttpPost("downloadattachments")]
-        [Produces(typeof(HttpResponseMessage))]
-        [SwaggerOperation("downloadAttachments")]
-        [SwaggerResponse(HttpStatusCode.OK, Description = "Returns IActionResult is of type HttpResponseMessage", Type = typeof(HttpResponseMessage))]
-        [SwaggerResponseRemoveDefaults]
-        public IActionResult DownloadAttachments([FromBody]MailAttachmentDetails mailAttachmentDetails)
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs]: Removed this method as it is not necessary

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs [hunk 5]
```diff
@@ -95,86 +102,35 @@ public IActionResult DownloadAttachmentsAsStream([FromBody]MailAttachmentDetails
...
                 throw;
             }
-        }
-
-        ///// <summary>
-        ///// Generates the stream from mail content.
-        ///// </summary>
-        ///// <param name="streamValue">The stream value.</param>
-        ///// <returns>Memory Stream.</returns>
-        private MemoryStream GenerateStreamFromString(string streamValue)
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs]: Removed this method as it is not necessary

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs [hunk 6]
```diff
@@ -95,86 +102,35 @@ public IActionResult DownloadAttachmentsAsStream([FromBody]MailAttachmentDetails
...
-                }
-                #endregion 
-                string emailText = WebUtility.UrlDecode(mailAttachmentDetails.AttachmentContent);
-                MemoryStream emailStream = GenerateStreamFromString(emailText);
-                string emailName = documentSettings.TempEmailName + new Guid().ToString() + ServiceConstants.EMAIL_FILE_EXTENSION;
-                HttpResponseMessage result = new HttpResponseMessage(HttpStatusCode.OK);                
-                result.Content = new StreamContent(emailStream);
-                result.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
-                result.Content.Headers.Add("content-disposition", "inline; filename=" + emailName);
+                fileContentResponse.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue(DocumentProvision.ReturnExtension(string.Empty));
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs]: HttpResponseMessage has been constructed and sending that HttpResponseMessage as a response. This HttpResponseMessage  contains complete attachments information that clients wants to download

File: tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.jobs/appSettings.json [hunk 7]
```diff
@@ -14,7 +14,8 @@
     "CentralRepositoryUrl": "",
     "AdminUserName": "",
     "AdminPassword": "",
-    "MatterUsersGroup": "Matter Center Users"
+    "MatterUsersGroup": "Matter Center Users",
+    "Tenant": "msmattercela.onmicrosoft.com"
```
Comment (WilsonReddy) [inline tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.jobs/appSettings.json]: don't check-in tenant value in future checkins.

**Pull request merged**